.circleci/config.yml

```diff
@@ -0,0 +1,118 @@
+version: 2.0
+
+jobs:
+  checkout:
+    docker:
+      - image: circleci/ruby:2-node-browsers
+    environment:
+      FLOW_CONTEXT=Production
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - yarn-cache-{{ checksum "yarn.lock" }}
+
+      - run: curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.34.0/install.sh | bash
+      - run: chmod +x ~/.nvm/nvm.sh
+      - run: ~/.nvm/nvm.sh install && ~/.nvm/nvm.sh use
+      - run: make install && make build-production
+
+      - save_cache:
+          key: yarn-cache-{{ checksum "yarn.lock" }}
+          paths:
+            - ~/.cache/yarn
+      - save_cache:
+          key: full-package-cache-{{ .Revision }}
+          paths:
+            - /home/circleci/project
+
+  codestyle:
+    docker:
+      - image: circleci/ruby:2-node-browsers
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - full-package-cache-{{ .Revision }}
+      - run: make lint
+
+  unittests:
+    docker:
+      - image: circleci/ruby:2-node-browsers
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - full-package-cache-{{ .Revision }}
+      - run: make test
+
+  build_flow_app:
+    docker:
+      - image: quay.io/yeebase/ci-build:7.2
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - full-app-cache2-{{ checksum "Build/CircleCi/composer.json" }}
+      - run: mkdir -p /home/circleci/app/Configuration
+      - run: cp Build/CircleCi/composer.* /home/circleci/app
+      - run: cp Build/CircleCi/Settings.yaml /home/circleci/app/Configuration/Settings.yaml
+      - run: cd /home/circleci/app && composer install
+      - save_cache:
+          key: full-app-cache2-{{ checksum "Build/CircleCi/composer.json" }}
+          paths:
+            -  /home/circleci/app
+
+  e2e:
+    environment:
+      FLOW_CONTEXT=Production
+    docker:
+      - image: quay.io/yeebase/ci-build:7.2
+      - image: circleci/mariadb:10.2
```

Comment:
What does it me that there are two images here @johannessteu?

Reply:
It's running 2 docker containers. The first one is the container where the steps are executed and the second is a mariadb that is running and can be used